### DISCUSSION TITLE
How to create a filter for table columns which have relationships?

### DISCUSSION THREAD

**Discussion by sergey-dme** [Budibase/budibase#1756] [Help] [answered]
Currently, Filtering in Data Provider or Repeater doesn't allow to work with/select table columns which have relationships with other tables. They are not showing in filter's column field. 
Is there a way to create such filter? (I'd like to select values in options pickers and get a table with results)

Thanks.

**Comment by mike12345567** [comment, marked as answer]
Hi @sergey-dme sadly this is currently a bit of a limitation of Budibase relationships, as they aren't stored in the row itself and are attached to search results on the way out of the system there is no way for us to integrate relationship contents onto the search capabilities.

In the future we will likely make this possible client-side, with the ability to transform/filter rows in Javascript, this is something that has been discussed.

**Reply by dswillem** [reply]
Is there already a solution for this? Filtering a table based on data from a relationship column or calculated column is not working?
Is this possible by javascript and how can we implement this?

**Reply by mjashanks** [reply]
Hi @dswillem. There is still no way to achieve this using Budibase internal tables.

**Comment by dswillem** [comment]
That’s really pitty. A simple database concept and not able to filter on columns with relationship and calculated fields.
